core/src/main/java/com/datastax/oss/driver/internal/core/metadata/MetadataManager.java

```diff
@@ -404,11 +415,7 @@ private void startSchemaRequest(CompletableFuture<Metadata> future) {
             .whenComplete(
                 (v, error) -> {
                   if (error != null) {
-                    Loggers.warnWithException(
-                        LOG,
-                        "[{}] Unexpected error while refreshing schema, skipping",
-                        logPrefix,
-                        error);
+                    currentSchemaRefresh.completeExceptionally(error);
```

Comment:
It's unlikely that an error will surface here, because the previous stages should have caught it. But it doesn't hurt to keep the check.

Reply:
Out of curiosity, what is the rationale for moving the log messages outside `MetadataManager`?

Reply:
It's because we don't log in all cases. In my opinion, an error should either be logged or surfaced through the user API, but not both.
For example:
* if the refresh is triggered by an incoming `SCHEMA_CHANGE` event on the control connection, there is no way to surface it so we log.
* but if the user called `session.refreshSchema`, we can rethrow the error and let them handle it. I don't like to log in those cases, because it will show an error in the logs even if the application already handled it.